app/models/agents/http_status_agent.rb

```diff
@@ -47,28 +50,39 @@ def validate_options
     end
 
     def check
-      check_this_url interpolated[:url]
+      check_this_url interpolated[:url], interpolated[:headers]
     end
 
     def receive(incoming_events)
       incoming_events.each do |event|
         interpolate_with(event) do
-          check_this_url interpolated[:url]
+          check_this_url interpolated[:url], interpolated[:headers]
         end
       end
     end
 
     private
 
-    def check_this_url(url)
+    def check_this_url(url, headers)
+      # Track time
       measured_result = TimeTracker.track { ping(url) }
+
+      payload = { 'url' => url, 'response_received' => false, 'elapsed_time' => measured_result.elapsed_time }
+
+      # Deal with failures
       if measured_result.result
-        create_event payload: { 'url' => url, 'status' => measured_result.status.to_s, 'response_received' => true, 'elapsed_time' => measured_result.elapsed_time }
+        payload.merge!({ 'response_received' => true, 'status' => measured_result.status.to_s })
+        # Deal with headers
+        if headers.present?
+          payload.merge!({ 'headers' => headers, 'header_values' => measured_result.result.headers[headers] })
```

Comment:
I think this format is pretty hard to process downstream, did you consider keeping the headers in a hash?